### PR TITLE
SPLICE-1290 Adding Batch Write Operations to Dictionary

### PR DESCRIPTION
**Pull request by jleach4**
Optimization for the dictionary to batch writes to main tables and indexes.

We behave a bit sluggishly when we are over a network partition due to all the RPC requests.

### CODE REVIEW

File: mem_sql/src/main/resources/log4j.properties [hunk 1]
```diff
@@ -18,9 +18,9 @@ log4j.appender.Console1.layout=org.apache.log4j.PatternLayout
 log4j.appender.Console1.layout.ConversionPattern=%d{HH:mm:ss,SSS} (%t) %-5p [%c] - %m%n
 
 log4j.rootLogger=INFO, Console1
-
-log4j.logger.com.splicemachine.derby.impl.store.access=INFO, Console1
-#log4j.logger.com.splicemachine.derby.impl.SpliceService=INFO, Console1
-log4j.logger.com.splicemachine.derby.utils=INFO, Console1
-log4j.logger.com.splicemachine.si.impl=INFO, Console1
-log4j.logger.com.splicemachine.db.impl.sql.catalog.DataDictionaryCache=INFO, Console1
```
Comment (dgomezferro) [inline mem_sql/src/main/resources/log4j.properties]: Did you mean to commit this?

File: splice_machine/src/main/java/com/splicemachine/derby/ddl/DDLWatchRefresher.java [hunk 2]
```diff
@@ -172,6 +172,7 @@ public boolean canUseCache(TransactionManager xact_mgr) {
 
     private void processPreCommitChange(DDLChange ddlChange,
                                         Collection<DDLWatcher.DDLListener> ddlListeners) throws StandardException {
+        LOG.error("processPreCommitChanges -> " + ddlChange);
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/ddl/DDLWatchRefresher.java]: This should be TRACE level

File: splice_machine/src/main/java/com/splicemachine/derby/ddl/DDLWatchRefresher.java [hunk 3]
```diff
@@ -191,6 +192,7 @@ private void clearFinishedChanges(Collection<String> children,Collection<DDLWatc
             if(!children.contains(entry)){
                 LOG.debug("Removing change with id " + entry);
                 changeTimeouts.remove(entry);
+                LOG.error("clearFinishedChanges -> " + entry);
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/ddl/DDLWatchRefresher.java]: Use TRACE/DEBUG here

File: splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/base/SpliceController.java [hunk 4]
```diff
@@ -221,6 +221,7 @@ public boolean batchFetch(List<RowLocation> locations, List<ExecRow> destRows,Fo
             }
             return true;
         }catch(Exception e){
+            e.printStackTrace();
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/base/SpliceController.java]: Remove or change to log

File: splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/btree/IndexController.java [hunk 5]
```diff
@@ -65,6 +69,51 @@ public IndexController(OpenSpliceConglomerate openSpliceConglomerate,
...
+            while (results.hasNext()) {
+                DataResult result = results.next();
+                if(result!=null && result.size()>0) {
+                    return ConglomerateController.ROWISDUPLICATE;
+                }
+            }
+            htable.writeBatch(puts.toArray(new DataPut[puts.size()]));
+            return 0;
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/btree/IndexController.java]: Log or remove

**Comment by dgomezferro** [comment]
Nice optimization!

**Comment by dgomezferro** [comment]
@jleach4 , would it be possible to implement the non-batch inserts in the *Controller classes in terms of the batch inserts to avoid code duplication? Or would it have a performance penalty?

**Comment by jleach4** [comment]
I like that idea, let me clean this up and re-submit.

**Pull request closed without merging**